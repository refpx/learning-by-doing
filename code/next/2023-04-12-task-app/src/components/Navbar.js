'use client'

import Link from 'next/link'
import { useRouter } from 'next/navigation'

function Navbar() {
  const router = useRouter()

  return (
    <header>
      <Link href='/'>
        <h1>My App</h1>
      </Link>

      <div>
        <button onClick={() => router.push('/new')}>Add task</button>
      </div>
    </header>
  )
}
export default Navbar
