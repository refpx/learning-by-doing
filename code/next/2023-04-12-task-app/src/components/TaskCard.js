import { useTasks } from '@/context/TasksContext'
import { useRouter } from 'next/navigation'
import { toast } from 'sonner'

function TaskCard({ task }) {
  const router = useRouter()
  const { deleteTask } = useTasks()

  return (
    <div
      style={{ background: '#ddd', color: '#000' }}
      onClick={() => router.push(`/edit/${task.id}`)}
    >
      <h2>{task.title}</h2>
      <button
        onClick={(e) => {
          e.stopPropagation()
          deleteTask(task.id)
          toast.success('Task deleted successfully')
        }}
      >
        Delete
      </button>
      <p>{task.description}</p>
    </div>
  )
}
export default TaskCard
