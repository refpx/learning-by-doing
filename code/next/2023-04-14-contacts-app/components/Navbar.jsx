import Link from "next/link"

function Navbar() {
  return (
    <div>
      <h1>Contacts App</h1>
      <nav>
        <ul>
          <li>
            <Link href="/">Home</Link>
          </li>
          <li>
            <Link href="/about">About</Link>
          </li>
          <li>
            <Link href="/services">Services</Link>
          </li>
        </ul>
      </nav>
    </div>
  )
}
export default Navbar