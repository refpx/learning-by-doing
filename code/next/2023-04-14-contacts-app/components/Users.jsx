/* eslint-disable @next/next/no-img-element */
'use client'
import { useRouter } from "next/navigation"

function Users({users}) {
  const router = useRouter()

  return (
    <ul className="grid grid-flow-row border border-gray-900 rounded-sm">
      {
        users.map(user => (
          <li
            key={user.id}
            onClick={() => {
              router.push(`/users/${user.id}`)
            }}
            className="flex items-center justify-between p-4 border-b border-gray-900 hover:bg-gray-700 hover:text-white cursor-pointer"
          >
            <div>
              <h5 className="font-bold">{user.id} {user.first_name} {user.last_name}</h5>
              <p>{user.email}</p>
            </div>
            <img 
              src={user.avatar} 
              alt={user.email} 
              className="w-20 h-20 rounded-full"
            />
          </li>
        ))
      }
    </ul>
  )
}
export default Users