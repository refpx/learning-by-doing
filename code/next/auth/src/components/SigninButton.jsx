'use client'

import { signIn, signOut, useSession } from 'next-auth/react'

const SigninButton = () => {
  const { data: session } = useSession()

  if (session && session.user) {
    return (
      <div className='flex gap-4 ml-auto'>
        <p className='text-sky-600 hover:text-sky-800'>{session.user.name}</p>
        <button onClick={() => signOut()}>Sign out</button>
      </div>
    )
  }

  return (
    <button
      onClick={() => signIn()}
      className='text-green-600 ml-auto hover:text-green-800'
    >
      Sign In
    </button>
  )
}
export default SigninButton
