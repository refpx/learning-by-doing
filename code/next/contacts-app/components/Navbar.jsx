import Link from "next/link"

function Navbar() {
  return (
    <div className="bg-gray-900 text-white">
      <nav className=" flex flex-wrap items-center justify-between mx-auto w-9/12 py-4">
      <h1 className="text-2xl font-bold">
        <Link href="/">Contacts App</Link>
      </h1>
      <ul className="flex flex-row gap-4">
        <li className="hover:underline">
          <Link href="/about">About</Link>
        </li>
        <li className="hover:underline">
          <Link href="/services">Services</Link>
        </li>
      </ul>
    </nav>
    </div>
  )
}
export default Navbar