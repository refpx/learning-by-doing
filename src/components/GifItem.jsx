export function GifItem({ url, title }) {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

