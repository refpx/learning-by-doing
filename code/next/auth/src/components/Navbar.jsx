import Link from 'next/link'
import SigninButton from './SigninButton'

const Navbar = () => {
  return (
    <header className='flex gap-4 p-4 bg-gradient-to-b from-white to-gray-200 shadow'>
      <SigninButton />
      <Link
        href='/users'
      >
        Users List
      </Link>
    </header>
  )
}
export default Navbar
