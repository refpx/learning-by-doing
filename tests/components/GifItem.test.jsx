import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem'

describe('Test en <GifItem />', () => {
  const title = 'Dota'
  const url = 'https://localhost/dota.jpg'

  test('debe de hacer match con el snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />)
    expect(container).toMatchSnapshot()
  })

  test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GifItem title={title} url={url} />)
    // screen.debug()
    // expect(screen.getByRole('img').alt).toBe(title)
    // expect(screen.getByRole('img').src).toBe(url)

    const { src, alt } = screen.getByRole('img')
    expect(alt).toBe(title)
    expect(src).toBe(url)
  })

  test('debe de mostrar el tÃ­tulo en el componente', () => {
    render(<GifItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()
  })

})