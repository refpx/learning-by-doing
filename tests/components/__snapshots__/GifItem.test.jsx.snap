// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dota"
      src="https://localhost/dota.jpg"
    />
    <p>
      Dota
    </p>
  </div>
</div>
`;
